lib/mini_racer/truffleruby.rb

```diff
@@ -0,0 +1,371 @@
+# frozen_string_literal: true
+
+module MiniRacer
+
+  class Context
+
+    class ExternalFunction
+      private
+
+      def notify_v8
+        name = @name.encode(::Encoding::UTF_8)
+        wrapped = lambda do |*args|
+          converted = @parent.send(:convert_js_to_ruby, args)
+          begin
+            result = @callback.call(*converted)
+          rescue Polyglot::ForeignException => e
+            e = RuntimeError.new(e.message)
+            e.set_backtrace(e.backtrace)
+            @parent.instance_variable_set(:@current_exception, e)
+            raise e
+          rescue => e
+            @parent.instance_variable_set(:@current_exception, e)
+            raise e
+          end
+          @parent.send(:convert_ruby_to_js, result)
+        end
+
+        if @parent_object.nil?
+          # set global name to proc
+          result = @parent.eval_in_context('this')
+          result[name] = wrapped
+        else
+          parent_object_eval = @parent_object_eval.encode(::Encoding::UTF_8)
+          begin
+            result = @parent.eval_in_context(parent_object_eval)
+          rescue Polyglot::ForeignException => e
+            raise ParseError, "Was expecting #{@parent_object} to be an object", e.backtrace
+          rescue => e
```

Comment:
since the handling is the same you can use
```ruby
rescue Polyglot::ForeignException, StandardError => e
```
or
```ruby
rescue Exception => e
```
if rescuing Exception is wanted (probably not here)

Reply:
I updated to `rescue Polyglot::ForeignException, StandardError => e`